crates/configuration/src/parachain.rs

```diff
@@ -892,6 +896,41 @@ impl<C: Context> ParachainConfigBuilder<WithAtLeastOneCollator, C> {
         }
     }
 
+    /// Add a new full node using a nested [`NodeConfigBuilder`].
+    /// The node will be configured as a full node (non-validator).
+    pub fn add_node(
```

Comment:
maybe we can rename this to `with_fullnode` so the name is very clear about the type of node.
wdyt?